tracing-subscriber/src/layer.rs

```diff
@@ -808,6 +808,92 @@ where
     }
 }
 
+impl<L, S> Layer<S> for Option<L>
+where
+    L: Layer<S>,
+    S: Subscriber,
+{
+    #[inline]
+    fn new_span(&self, attrs: &span::Attributes<'_>, id: &span::Id, ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.new_span(attrs, id, ctx)
+        }
+    }
+
+    #[inline]
+    fn register_callsite(&self, metadata: &'static Metadata<'static>) -> Interest {
+        match self {
+            Some(ref inner) => inner.register_callsite(metadata),
+            None => Interest::always(),
+        }
+    }
+
+    #[inline]
+    fn enabled(&self, metadata: &Metadata<'_>, ctx: Context<'_, S>) -> bool {
+        match self {
+            Some(ref inner) => inner.enabled(metadata, ctx),
+            None => true,
+        }
+    }
+
+    #[inline]
+    fn max_level_hint(&self) -> Option<LevelFilter> {
+        match self {
+            Some(ref inner) => inner.max_level_hint(),
+            None => None,
+        }
```

Comment:
Nit, take it or leave it: this could also be expressed using [`Option::and_then`](https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then), like this: 
```suggestion
        self.and_then(Layer::max_level_hint)
```